### PR TITLE
<username>-che as default namespace

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
**Referenced issue**
issue: https://github.com/eclipse/che/issues/15493

**What does this PR do?**
Continue work started https://github.com/eclipse/che-operator/pull/137 :
 - don't create che-workspace SA
 - check what namespace we're using for workspaces (`workspaceNamespaceDefault`)
   - if it is different than che-operator is deployed to, create all ClusterRoles and ClusterRoleBindings needed to create new namespaces and to create workspaces in them
   - if it is same as che-operator's namespace, create just Role and RoleBinding needed to manage workspaces in same namespace
 - if oauth is enabled, just create `che` sa with no granted permissions
 - finalize cluster roles and cluster role bindings created by che-operator for Che server
 - make working Che server property CHE_INFRA_KUBERNETES_CLUSTER__ROLE__NAME

**Related pr(s)**
https://github.com/che-incubator/chectl/pull/469

TODO:
 - [x] test and fix permissions for k8s (currently it works for openshift)
 - [x] test with OS Oauth. PR does not grant any permissions for `che` sa
 - [x] check chectl
 - [ ] test installation with olm

### CODE REVIEW

File: pkg/deploy/che_configmap.go [hunk 1]
```diff
@@ -164,6 +166,7 @@ func GetConfigMapData(cr *orgv1.CheCluster) (cheEnv map[string]string) {
 		CheDebugServer:                       cheDebug,
 		CheInfrastructureActive:              infra,
 		CheInfraKubernetesServiceAccountName: "che-workspace",
+		CheWorkspaceClusterRole:			  cheWorkspaceClusterRole,
```
Comment (sparkoo) [inline pkg/deploy/che_configmap.go]: formatting looks bit off here

**Comment by sparkoo** [comment]
When used with chectl, I'm getting:
```
E0128 07:19:57.695018       1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:196: Failed to list *v1.ClusterRoleBinding: clusterrolebindings.rbac.authorization.k8s.io is forbidden: User "system:serviceaccount:che:che-operator" cannot list resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope
```

**Comment by sparkoo** [comment]
It looks like the patch is not actually changing default namespace. It has to be defined in `CheCluster` CR yaml, whitch is now empty https://github.com/eclipse/che-operator/blob/pr/137/deploy/crds/org_v1_che_cr.yaml#L57
It can be done as separate PR. This PR will make it ready for it.

**Comment by AndrienkoAleksandr** [comment]
> currently does not work with chectl 

pr for chectl https://github.com/che-incubator/chectl/pull/469 . @sparkoo Did you tested this one?

**Comment by sparkoo** [comment]
> > currently does not work with chectl
> 
> pr for chectl [che-incubator/chectl#469](https://github.com/che-incubator/chectl/pull/469) . @sparkoo Did you tested this one?

no :) will do

**Comment by sparkoo** [comment]
works with https://github.com/che-incubator/chectl/pull/469. Just need to properly handle templates.

**Comment by AndrienkoAleksandr** [comment]
Hello, @davidfestal, @tolusha , @mmorhun . I completed testing current pr with OLM on the minikube and CRC clutster. Looks like everything works just fine. Could you review pr, please?

File: pkg/controller/che/che_controller.go [hunk 2]
```diff
@@ -1191,6 +1155,179 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
...
+					Resources: []string{"namespaces"},
+					Verbs: []string{"get"},
+				},
+			})
+			if err := r.CreateNewClusterRole(instance, cheManageNameSpacesRole); err != nil {
+				return err
+			}
+
+			// this binding is needed to manage che workspaces out of che namespace
+			// `${namespace}-clusterrole-manage-namespaces` ClusterRole should be created during che-operator deploy
```
Comment (sparkoo) [inline pkg/controller/che/che_controller.go]: If ClusterRole is created during che-operator deploy, like this comment say, why it is created again here https://github.com/eclipse/che-operator/pull/156/files#diff-d9445e614f0d728fd140771b0fd29c50R1315 ?

Comment (AndrienkoAleksandr) [reply pkg/controller/che/che_controller.go]: OLM auto-generate unique names for che-operator ClusterRoles. ClusterRole object is not namespaced it has wider scope. That's why olm generate name to have ability to use few operators in the same cluster with own specific version of the ClusterRoles and their rules. We can not find and reuse this roles for che-server by name, that why I created for che-server own Cluster roles. 
P.S.: Theoretically we can reuse che-operator cluster role using labels, but I used current approach.

Comment (sparkoo) [reply pkg/controller/che/che_controller.go]: I see, ok. thanks

File: README.md [hunk 3]
```diff
@@ -131,6 +131,7 @@ Or you can build in a container:
 docker run -ti -v /tmp:/tmp -v ${OPERATOR_REPO}:/opt/app-root/src/go/src/github.com/eclipse/che-operator registry.redhat.io/devtools/go-toolset-rhel7:1.11.5-3 sh -c "OOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o /tmp/run-tests /opt/app-root/src/go/src/github.com/eclipse/che-operator/e2e/*.go"
 cp /tmp/run-tests ${OPERATOR_REPO}/run-tests
 ```
+> Notice: if you don't have redhat subscription, use public image registry.access.redhat.com/devtools/go-toolset-rhel7:latest
```
Comment (mmorhun) [inline README.md]: I think it is not right that we have link for private images in Che project. But this is out of scope of this PR.

**Comment by AndrienkoAleksandr** [comment]
Sorry guys, pr moved https://github.com/eclipse/che-operator/pull/166, because Ci become crazy with branch name pr/137 ... And github doesn't provide ability to rename target branch for pr. I can change base branch, but not target.
